File: golden-paths/php/slim/tests/DeliberateTest.php
<?php
// Deliberately-failing test — proves the php lane PROPAGATES failure. It carries the PHPUnit
// `selfcheck` group; a normal run uses --exclude-group selfcheck and the lane self-check uses
// --group selfcheck (only it). Group-based (not a test-NAME filter): dropping the group makes it run
// in the NORMAL lane, which is loud, not silent.
declare(strict_types=1);

use PHPUnit\Framework\Attributes\Group;
use PHPUnit\Framework\TestCase;

final class DeliberateTest extends TestCase
{
    #[Group('selfcheck')]
    public function testDeliberateFailure(): void
    {
        self::fail('selfcheck: the golden-path php lane must surface this failure (exit non-zero)');
    }
}

File: golden-paths/ruby/sinatra/selfcheck/deliberate_test.rb
# frozen_string_literal: true

# Deliberately-failing test — proves the ruby lane PROPAGATES failure. It lives in selfcheck/, OUTSIDE
# the `test` rake task's test/**/ glob, so a normal run never sees it; `rake test:selfcheck` runs ONLY
# it. Standalone (no app boot) so the self-check is fast and independent.
require "minitest/autorun"

class DeliberateTest < Minitest::Test
  def test_deliberate_failure
    flunk "selfcheck: the golden-path ruby lane must surface this failure (exit non-zero)"
  end
end
